### PR TITLE
add script to clean Nexus3 docker registry

### PR DESCRIPTION
**Pull request by jabielecki**

### CODE REVIEW

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    def retentionDate = DateTime.now().minusHours(repoToClean.retentionHours);
+    log.info(":::Cleanup script started for ${repositoryName} with the retention date ${retentionDate}");
+
+    MaintenanceService service = container.lookup("org.sonatype.nexus.repository.maintenance.MaintenanceService");
+    def repo = repository.repositoryManager.get(repositoryName);
+    def tx = repo.facet(StorageFacet.class).txSupplier().get();
+    try {
+        tx.begin();
+        def assets = tx.browseAssets(tx.findBucket(repo));
+        tx.rollback()
```
Comment (kravvcuCL) [inline tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:52]: why a rollback here?

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        int deletedComponentCount = 0
+        int skippedComponentCount = 0
+        def keepingComponents = [:];
+        ImmutableList.copyOf(assets).each{ asset ->
+            if( asset.componentId() != null ) {
+                def ch = asset.attributes().child("content");
+                //log.info("ch=${ch}");
```
Comment (kravvcuCL) [inline tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:59]: Remove if not needed

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import com.google.common.collect.ImmutableList;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.maintenance.MaintenanceService;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.docker.DockerGCFacet ;
+import org.sonatype.nexus.repository.docker.internal.DockerGCFacetImpl ;
+import static org.sonatype.nexus.common.time.DateHelper.toDateTime;
+
+
+for( repoToClean in repositoriesToClean ) {
```
Comment (kravvcuCL) [inline tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy]: spaces unneeded aroun the expression

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                        }
+                    }
+                }
+            }
+        }
+        log.info("Deleted ${deletedComponentCount} docker image-tags from ${repositoryName}");
+        log.info("Keeping ${keepingComponents.size()} image-tags which had recent assets");
+        log.info("Skipped ${skippedComponentCount} recent image-tags which had old assets");
+
+        //ImmutableList.copyOf(components).each { comp ->
```
Comment (kravvcuCL) [inline tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:91]: delete if not needed

**Review by kravvcuCL — CHANGES_REQUESTED** [review]

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy (hunk 1 above)
Comment (jabielecki) [reply tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:52]: Doesn't work without begin, so it needs either rollback or commit. Rollback might be easier on db than a commit.

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy (hunk 2 above)
Comment (jabielecki) [reply tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:59]: It's not a mature code, I actually left these comments on purpose. There's a good chance we need these to debug production soon.

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy (hunk 4 above)
Comment (jabielecki) [reply tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy:91]: It's not a mature code, there is a good chance this fragment will be needed - where do I keep this fragment?

File: tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy (hunk 3 above)
Comment (jabielecki) [reply tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy]: Done here and everywhere - thanks for spotting.

**Review by kravvcuCL — APPROVED** [review]

**Pull request merged**